pkg/slo-controller/noderesource/plugins/midresource/plugin.go

```diff
@@ -163,18 +172,34 @@ func (p *Plugin) calculate(strategy *configuration.ColocationStrategy, node *cor
 		allocatableMemory = 0
 	} else {
 		prodReclaimable := resourceMetrics.NodeMetric.Status.ProdReclaimableMetric.Resource
-		allocatableMilliCPU = prodReclaimable.Cpu().MilliValue()
-		allocatableMemory = prodReclaimable.Memory().Value()
-		prodReclaimableMilliCPU = allocatableMilliCPU
-		prodReclaimableMemory = prodReclaimable.Memory().String()
+		prodReclaimableCPU = prodReclaimable.Cpu()
+		prodReclaimableMemory = prodReclaimable.Memory()
+		allocatableMilliCPU = prodReclaimableCPU.MilliValue()
+		allocatableMemory = prodReclaimableMemory.Value()
 	}
 
-	nodeAllocatable := node.Status.Allocatable
+	nodeMetric := resourceMetrics.NodeMetric
+
+	hostAppHPUsed := resutil.GetHostAppHPUsed(resourceMetrics)
```

Comment:
To calculate the allocatable of the Mid-tier, it should ignore the host used of the Mid priority.

Reply:
fixed